Build fails with git configured to show signatures

## I'm submitting a…
<!-- Please check one of the following options with "x" -->
<pre>
[x] Bug
[ ] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior

Running the configure script fails when git is configured to show commit signatures (`log.showSignature=true`).

## Expected Behavior

i3 should not fail to build based on my git configuration.

## Reproduction Instructions

- set the git config variable `log.showSignature` to `true`, e.g. in `~/.config/git/config`:

    ```ini
    [log]
    showSignature = true
    ```
- `mkdir build && cd build && ../configure`

<details><summary>The interesting bit of <code>config.log</code></summary>

```
configure:4407: checking whether the C compiler works
configure:4429: gcc    conftest.c  >&5
conftest.c:10:20: warning: missing terminating " character
 #define I3_VERSION "4.16-97-g3aa42cf79 (gpg: Signature made Thu Mar 21 07:44:13 2019 GMT
                    ^
conftest.c:11:4: error: expected '=', ',', ';', 'asm' or '__attribute__' before ':' token
 gpg:                using RSA key 4AEE18F83AFDEB23
    ^
conftest.c:11:35: error: invalid suffix "AEE18F83AFDEB23" on integer constant
 gpg:                using RSA key 4AEE18F83AFDEB23
                                   ^~~~~~~~~~~~~~~~
conftest.c:12:9: warning: missing terminating ' character
 gpg: Can't check signature: No public key
         ^
conftest.c:12:9: error: missing terminating ' character
 gpg: Can't check signature: No public key
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
conftest.c:13:20: error: stray '\' in program
 2019-03-21, branch \"next\")"
                    ^
configure:4433: $? = 1
configure:4471: result: no
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "i3"
| #define PACKAGE_TARNAME "i3"
| #define PACKAGE_VERSION "4.16"
| #define PACKAGE_STRING "i3 4.16"
| #define PACKAGE_BUGREPORT "https://github.com/i3/i3/issues"
| #define PACKAGE_URL ""
| #define PACKAGE "i3"
| #define VERSION "4.16"
| #define I3_VERSION "4.16-97-g3aa42cf79 (gpg: Signature made Thu Mar 21 07:44:13 2019 GMT
| gpg:                using RSA key 4AEE18F83AFDEB23
| gpg: Can't check signature: No public key
| 2019-03-21, branch \"next\")"
| #define MAJOR_VERSION 4
| #define MINOR_VERSION 16
| #define PATCH_VERSION 0
| #define UNUSED_NDEBUG /**/
| /* end confdefs.h.  */
|
| int
| main ()
| {
|
|   ;
|   return 0;
| }
configure:4476: error: in `/home/josh/src/i3/build':
configure:4478: error: C compiler cannot create executables
See `config.log' for more details
```

</details>

I think I've run into this before – the problem is on line 35 of `configure.ac`, with this call to git:

    git -C ${srcdir} log --pretty=format:%cd --date=short -n1

`git log` is a porcelain command, so the proper solution is to not use it in scripts, but as a hack to make it work in this specific case, you can add the `--no-show-signature` flag.

## Environment
<!--
Please include your exact i3 version.
Note that we only support the latest major release and the current development version. If you are using an older version of i3, please first update to the current release version and reproduce the issue there.
-->
Output of `i3 --moreversion 2>&-`:
<pre>
Binary i3 version:  4.16.1 (2019-01-27, branch "tags/4.16.1") © 2009 Michael Stapelberg and contributors
Running i3 version: 4.16.1 (2019-01-27, branch "tags/4.16.1") (pid 1845)
Loaded i3 config: /home/josh/.config/i3/config (Last modified: Thu 21 Mar 2019 21:02:02 GMT, 330 seconds ago)

The i3 binary you just called: /usr/local/bin/i3
The i3 binary you are running: i3
</pre>

<!--
Please also answer the questions below to help us process your issue faster. If you have any other information to share, please add it here as well.
-->
<pre>
- Linux Distribution & Version: Ubuntu 18.04
- Are you using a compositor (e.g., xcompmgr or compton):
</pre>

--- a/configure.ac
+++ b/configure.ac
@@ -32,7 +32,7 @@ AX_EXTEND_SRCDIR
 AS_IF([test -d ${srcdir}/.git],
       [
         VERSION="$(git -C ${srcdir} describe --tags --abbrev=0)"
-        I3_VERSION="$(git -C ${srcdir} describe --tags --always) ($(git -C ${srcdir} log --pretty=format:%cd --date=short -n1), branch \\\"$(git -C ${srcdir} describe --tags --always --all | sed s:heads/::)\\\")"
+        I3_VERSION="$(git -C ${srcdir} describe --tags --always) ($(git -C ${srcdir} rev-list --format=%cd --date=short -n1 $(git rev-parse HEAD) | tail -n1), branch \\\"$(git -C ${srcdir} describe --tags --always --all | sed s:heads/::)\\\")"
         # Mirrors what libi3/is_debug_build.c does:
         is_release=$(test $(echo "${I3_VERSION}" | cut -d '(' -f 1 | wc -m) -lt 10 && echo yes || echo no)
       ],

